### PR TITLE
Added scanning sonar parameters.

### PR DESCRIPTION
**Pull request by romulogcerqueira**
* Created getAdInterval() function to calculate the sample time period;
* Initialized scanning sonar parameters (time, sampling_interval, speed_of_sound, beamwidth_horizontal and beamwidth_vertical);
* Created range and step_angle variables.

### CODE REVIEW

File: src/ScanSonar.cpp [hunk 1]
```diff
@@ -68,47 +68,58 @@ std::vector<uint8_t> ScanSonar::getPingData(cv::Mat raw_intensity) {
 	return ping_intensity;
 }
 
+// Calculate the sample time period that is applied to the received Sonar echo signal
+uint16_t ScanSonar::getADInterval() {
```
Comment (doudou) [inline src/ScanSonar.cpp]: This is a micron-specific conversion. Use a generic SI-based unit at this level of the codebase.

File: src/ScanSonar.cpp [hunk 2]
```diff
@@ -68,47 +68,58 @@ std::vector<uint8_t> ScanSonar::getPingData(cv::Mat raw_intensity) {
...
+	float travel_time = _range * 2.0 / _speed_of_sound;
+	float sample_time = travel_time / (float) _number_of_bins;
+	uint8_t ad_interval = (uint8_t) ((sample_time / (640 * 1e-9)) + 0.5);
```
Comment (doudou) [inline src/ScanSonar.cpp]: Plus, you calculate with uint8_t and return uint16_t ...

File: src/ScanSonar.cpp (hunk 1 above)
Comment (romulogcerqueira) [inline src/ScanSonar.cpp]: Hi Sylvain,

I'm not sure if I understood correctly. Is ad_interval parameter a micron-specific configuration or the way to calculate the sampling interval?

File: src/ScanSonar.cpp (hunk 2 above)
Comment (romulogcerqueira) [inline src/ScanSonar.cpp]: Fixed! I will commit the updated source code soon.

**Comment by romulogcerqueira** [comment]
@doudou updated code.

File: src/ScanSonar.cpp (hunk 1 above)
Comment (doudou) [inline src/ScanSonar.cpp]: The way to calculate. Make that method return something that has sane units (e.g. sample_time directly). The micron-specific conversion to a weird unit should not end up in the simulation code.

File: test/SonarSim_test.cpp [hunk 3]
```diff
@@ -82,6 +77,36 @@ void makeManifold(osg::ref_ptr<osg::Group> root) {
 	root->addChild(ptransform);
 }
 
+//draw the scene with a small ball in the center with a big cube, Box and cone in back
+void addFlatfish(osg::ref_ptr<osg::Group> root) {
+
+	osg::Node* flatfish = osgDB::readNodeFile("/home/romulogc/rock_robotics/simulation/orogen/gpu_sonar_simulation/resources/flatfish_02.dae");
```
Comment (doudou) [inline test/SonarSim_test.cpp]: don't hardcode full paths. In principle, readNodeFile looks in all directories included in the OSG_FILE_PATH environment variable, so use this instead.

Comment (romulogcerqueira) [inline test/SonarSim_test.cpp]: I will correct it and commit soon. Thanks!

Comment (romulogcerqueira) [inline test/SonarSim_test.cpp]: updated sampling interval.

Comment (doudou) [inline test/SonarSim_test.cpp]: ping ... full path still there

**Comment by romulogcerqueira** [comment]
Sylvain,

sorry for my delay to answer you. During Marco's vacations, I have assumed some administrative tasks.

I corrected the test source code and committed it.

**Comment by doudou** [comment]
> sorry for my delay to answer you

No problem.

**Pull request merged**